src/array/dictionary/mutable.rs

```diff
@@ -63,6 +63,22 @@ impl<K: DictionaryKey, M: MutableArray + Default> Default for MutableDictionaryA
 }
 
 impl<K: DictionaryKey, M: MutableArray> MutableDictionaryArray<K, M> {
+    /// Creates an [`MutableDictionaryArray`] from a given keys array and values array.
+    /// # Errors
+    /// Errors if the keys array have duplicates or if the keys are out of bounds of the values array.
+    pub fn try_new(data_type: DataType, keys: MutablePrimitiveArray<K>, values: M) -> Result<Self>
+    where
+        M: Indexable,
+        M::Type: Eq + Hash,
+    {
+        let map = ValueMap::<K, M>::from_values(values)?;
+        Ok(Self {
```

Comment:
Hm... shouldn't `data_type` be validated here?

Reply:
No, I think what you need to check is that ValueMap's resulting data type is the same as data type being passed in?